cmd/clusterctl/examples/vsphere/cluster.yaml.template

```diff
@@ -17,3 +17,5 @@ spec:
       vspherePassword: "${VSPHERE_PASSWORD}"
       vsphereServer: "${VSPHERE_SERVER}"
       vsphereCredentialSecret: ""
+      sshAuthorizedKeys:
+      - "${SSH_AUTHORIZED_KEY}"
```

Comment:
Tested that an empty string here doesn't cause any issues, though it may still add unnecessary fields in cloud init.

Reply:
Hi @andrewsykim,

YAML can accept a single line array. Maybe templatize this as:

```
sshAuthorizedKeys: "${SSH_AUTHORIZED_KEY}" 
```

And in the shell script ensure that `SSH_AUTHORIZED_KEY` is set to `[ "${SSH_AUTHORIZED_KEY}" ]` whenever the env var is actually set.